openpype/hosts/maya/api/plugin.py

```diff
@@ -143,14 +143,50 @@ def load(
         assert os.path.exists(self.fname), "%s does not exist." % self.fname
 
         asset = context['asset']
+        subset = context['subset']
+        settings = get_project_settings(context['project']['name'])
+        custom_naming = settings['maya']['load']['reference_loader']['naming']
         loaded_containers = []
 
         count = options.get("count") or 1
+
         for c in range(0, count):
-            namespace = namespace or lib.unique_namespace(
-                "{}_{}_".format(asset["name"], context["subset"]["name"]),
-                prefix="_" if asset["name"][0].isdigit() else "",
-                suffix="_",
+            group_name = None
+
+            if ':' not in custom_naming:
+                raise ValueError(
+                    "Wrong format for namespace, missing ':' separator"
+                )
+            elif custom_naming.strip()[0] == ':':
+                raise ValueError(
+                    "Wrong format for namespace, missing content before ':' "
+                    "separator"
+                )
+            elif custom_naming.strip()[-1] == ':':
+                raise ValueError(
+                    "Wrong format for namespace, missing content after ':' "
+                    "separator"
+                )
```

Comment:
Can we move this logic out of the for loop? This check can be done much higher already - nearer to where we retrieve the setting.